contrib_bots/lib/john.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+import time
+
+from random import choice
+
+try:
+    from chatterbot import ChatBot
+    from chatterbot.trainers import ChatterBotCorpusTrainer, ListTrainer
+except ImportError:
+    raise ImportError("""It looks like you are missing chatterbot.
+                      Please: pip install chatterbot""")
+
+JOKES_PATH = os.path.expanduser('lib/John/jokes.json')
```

Comment:
We should use paths starting with "lib/John/var" here, where the "var" helps keep these files out of our Zulip checkout.  We use "var" as the conventional directory in Zulip for little throwaway files.

Also, we want to be careful that the paths here are correct here relative to os.path.dirname(os.path.abspath(__filename__)), so that we're not dropping this directory in a strange place.

We use this technique a lot in Zulip--see tools/test-backend for an example.